HealthAPILibrary/GiniHealthAPILibraryPinning/Sources/GiniHealthAPILibraryPinning/SSLPinning/SSLPinningManager.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  SSLPinningManager.swift
+//
+//
+//  Copyright © 2024 Gini GmbH. All rights reserved.
+//
+
+import CryptoKit
+import Foundation
+import CommonCrypto
+
+struct SSLPinningManager {
```

Comment:
[Clarification] The reason we used TrustKit is to not have to implement and maintain our own pinning implementation. If we go with this solution, then we are responsible of maintaining it and we are not familiar with the `CryptoKit` and `CommonCrypto` frameworks.

I am afraid that after you leave we will be very reluctant to touch this code. Even if you add detailed inline documentation for everything (for ex. what is `rsa2048ASN1Header` and why it's needed) we will have to acquire the level of knowledge you have before we can touch this code...

@zladzeyka my opinion is that we should stick with TrustKit and check if we can update to [version 3](https://github.com/datatheorem/TrustKit/releases). They are actively maintaining it. Otherwise we will have to learn how this code works and keep it updated and also be aware of any security leaks and address them, etc...

Reply:
regarding CommonCrypto - it's standard ios framework we use it only for getting SHA256 hash (needed for iOS 12, and can be removed soon) 
regarding CryptoKit - it's standard ios framework. So all responsibility on Apple :) 

we provide to TrustKit only hashed public keys and domains - so there is no other logic here possible - only check equality of keys hashes and domains. (I will add code for domain check ASAP) 

To be sure that everything works properly - we have tests. 
From my side - contact me anytime about my code, I'll be happy to help

Reply:
regarding rsa2048ASN1Header - it's standard too TrustKit hardcoded it too

https://github.com/datatheorem/TrustKit/blob/master/TrustKit/Pinning/TSKSPKIHashCache.m